src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy

```diff
@@ -100,28 +121,70 @@ class HaFlowExtended {
+            HaSubFlowExtended.builder()
+                    .haSubFlowId(it.flowId)
+                    .endpointSwitchId(new SwitchId(it.endpoint.switchId.id))
+                    .endpointPort(it.endpoint.portNumber)
+                    .endpointVlan(it.endpoint.vlanId)
+                    .endpointInnerVlan(it.endpoint.innerVlanId)
+                    .description(it.description)
+                    .status(FlowStatus.valueOf(it.status.toUpperCase().replace(" ", "_")))
+                    .build()
+        }
         this.timeCreate = haFlow.timeCreate
         this.timeUpdate = haFlow.timeUpdate
 
         this.northboundV2 = northboundV2
         this.topologyDefinition = topologyDefinition
+        this.cleanupManager = cleanupManager
+    }
+
+    HaFlowExtended create(FlowState expectedState = FlowState.UP, CleanupAfter cleanupAfter = TEST) {
+        cleanupManager.addAction(DELETE_HAFLOW, { delete() }, cleanupAfter)
+        def haFlow = northboundV2.addHaFlow(HaFlowCreatePayload.builder()
+                .haFlowId(this.haFlowId)
+                .sharedEndpoint(this.sharedEndpoint)
+                .maximumBandwidth(this.maximumBandwidth)
+                .pathComputationStrategy(this.pathComputationStrategy.toString())
+                .encapsulationType(this.encapsulationType.toString())
+                .maxLatency(this.maxLatency)
+                .maxLatencyTier2(this.maxLatencyTier2)
+                .ignoreBandwidth(this.ignoreBandwidth)
+                .periodicPings(this.periodicPings)
+                .pinned(this.pinned)
+                .priority(this.priority)
+                .strictBandwidth(this.strictBandwidth)
+                .description(this.description)
+                .allocateProtectedPath(this.allocateProtectedPath)
+        /*A dirty hack to not have 'diverse_flow_id' field in the object (it presents only in create request).
+        So we keep one string value added by HaFlowBuilder in 'diverseWithFlows' field, but it's overwritten
+        when we create/update HaFlow (based on values in server response which are more interesting)*/
+                .diverseFlowId(diverseWithFlows ? diverseWithFlows.first() : null)
+                .subFlows(subFlows.collect {
+                    HaSubFlowCreatePayload.builder()
+                            .endpoint(BaseFlowEndpointV2.builder()
+                                    .switchId(it.endpointSwitchId)
+                                    .portNumber(it.endpointPort)
+                                    .vlanId(it.endpointVlan)
+                                    .innerVlanId(it.endpointInnerVlan)
+                                    .build())
+                            .description(it.description)
+                            .build()
+                })
+                .build())
+        waitForBeingInState(expectedState)
+        return new HaFlowExtended(haFlow, northboundV2, topologyDefinition, cleanupManager)
```

Comment:
In the scope of flow creation, we wait for an ha-flow to be in UP state, but method returns HaFlowExtended from northbound response(flow is in_progress state).

Reply:
Fixed. Now `create()` returns what `waitForBeingInState()` returns.